### PR TITLE
Help for upgrading from v0.2.x to v0.3.0

### LINKED ISSUE #222: Add docs for migrating from bloomberg (v0.2.8) to apache (v0.3.0)
Migrating from v0.2.8 to v0.3.0 will require manual intervention, because the resource groups are changing.

Documentation is needed to make this transition as smooth as possible for users.

### PR DESCRIPTION
**Pull request by HoustonPutman**
Resolves #222 

This includes a test that installs v0.2.6, upgrades to v0.2.8 and then upgrades to v0.3.0.

The documentation gives two options, upgrading in an environment managed manually, and an environment managed via a system like Ansible.

### CODE REVIEW

File: hack/test-upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+####
+##  Install the Apache Solr Operator separately
+####
+# Build the Apache release (TODO: Change to release when it is cut)
+make docker-build
+helm dependency build helm/solr-operator
+
+# Uninstall Old ZK Operator
+echo "Deleting all Zookeeper Operator resources, except for CRDs" && \
+  kubectl delete deployment zk-operator && \
```
Comment (thelabdude) [inline hack/test-upgrade.sh:123]: for me the deployment is named `zookeeper-operator` vs. `zk-operator`

Comment (HoustonPutman) [reply hack/test-upgrade.sh:123]: How were you installing it for Solr Operator v0.2.x?

Comment (thelabdude) [reply hack/test-upgrade.sh:123]: I just did it independently of the solr-operator:
```
helm upgrade --install zookeeper-operator pravega/zookeeper-operator \
  --version=0.2.9 --namespace=${NAMESPACE}
```

Comment (HoustonPutman) [reply hack/test-upgrade.sh:123]: Ahhh if you are running it 0.2.9 already, you should just use `--set zookeeper-operator.install=false --set zookeeper-operator.use=true`

Probably good to add to the docs.

**Comment by thelabdude** [comment]
Should this PR change the chart version in `helm/solr-operator/Chart.yaml` to `0.3.0`? Cool if not ... just confused me when I installed the chart from a checkout of this PR and got:
```
helm ls
NAME              	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART                   	APP VERSION
apache            	tjp      	1       	2021-03-24 11:07:21.514488 -0600 MDT	deployed	solr-operator-0.2.8     	v0.2.8     
solr-operator     	tjp      	1       	2021-03-24 11:04:56.128582 -0600 MDT	deployed	solr-operator-0.2.8     	v0.2.8     
zookeeper-operator	tjp      	1       	2021-03-24 10:54:55.174334 -0600 MDT	deployed	zookeeper-operator-0.2.9	0.2.9      
```
If that's done as part of another PR or only during the release, please disregard ^^

**Comment by HoustonPutman** [comment]
> Should this PR change the chart version in helm/solr-operator/Chart.yaml to 0.3.0?

Yes. We should probably treat the version as the same as Solr, where what's in the branch represents the next release.

Will put a different PR for that.

**Review by thelabdude — APPROVED** [review]
This looks great! Very thorough. I was able to upgrade from 0.2.8 -> a dev build from this PR on GKE.

**Pull request merged**